crypto/BRCryptoWalletManagerClient.c

```diff
@@ -441,28 +540,54 @@ cwmTransactionEventAsBTC (BRWalletManagerClientContext context,
                                                      });
             }
 
+            cryptoTransferGive (transfer);
             break;
         }
 
         case BITCOIN_TRANSACTION_UPDATED: {
+            // This event occurs when the timestamp and/or blockHeight have been changed.
+            // Regardless of if this was a user created transaction or if it came from a
+            // sync, we are holding a copy of the transaction that BRWalletManager has.
+            // As a result, we need to update our copy of the transaction.
```

Comment:
Maybe a note in the comment about if there is a blockchain reorg we'll get an UPDATE with a block height of TX_UNCONFIRMED (which I see is handled below).

Reply:
Will update.